include/deal.II/base/parameter_handler.h

```diff
@@ -1024,13 +1024,21 @@ class ParameterHandler : public Subscriptor
   virtual void
   parse_input_from_json(std::istream &input, const bool skip_undefined = false);
 
+  /**
+   * Create a default input file by calling print_parameters() depending on the
+   * type of input file specified (prm, xml, json).
+   */
+  void
+  create_default_input_file(const std::string &filename,
```

Comment:
This should take an `OuptutStyle` instead of bool (see #9888, and #9919)

Reply:
clearly not. `OutputStyle` contains more information than the info of these function parameters. You are now actually blowing up the interface, against your promises in #9888.

Reply:
Why do you think that replacing a function that takes three parameters (two of which are boolean), with one that takes only two is "blowing up the interface"?
The user interface is going to be simpler, and will allow for any other options we may implement in the future without changes in the interface, but only in the (internal) implementation.

Reply:
Ok, I am willing to give more detailed explanations, but let's please stop this black-or-white type argumentations and all these rhetorical questions. 

It is not just counting parameters. A good interface reduces dependencies to a minimum (and also avoids redundancy). It's easy to put everything into a larger data structure and "reduce the interface" in your understanding by just plugging in this bigger data structure. When you make changes to the bigger data structure, you will also have to change many of the functions relying on this data structure. You claim to know exactly where changes will be made in the future and where not? That is part of the problem.

Example: Consider the function `Algorithms::Newton::declare_parameters()/parse_parameters()`. 

https://www.dealii.org/developer/doxygen/deal.II/classAlgorithms_1_1Newton.html

According to your definition, the interface is maximally lean and well-designed. I claim the opposite. I think you will agree that your favorite methods for parsing parameters differ from those used in ``Algorithms::Newton`` (namely ``add_parameter()`` vs. ``declare_entry()``). Furthermore, the interfaces of these function have been extended in ``ParameterHandler`` recently. It does not make sense that we have do update ``Algorithms::Newton`` when making changes to ``ParameterHandler``. In fact, we **do not have to** change ``Algorithms::Newton``, because we always try to be backwards compatible. But these are dependencies that lead to the fact that other modules will become outdated and making changes becomes more and more difficult, but these modules are actually completely independent.  Next, you might argue that you do not want to change the Newton solver whenever you add a new parameter for the Newton solver, sure. For this reason, I would introduce a data structure ``NewtonSolverData`` that is going to be extended whenever the functionality of the Newton solver is updated/changed. Imagine that whenever possible classes in deal.ii would use ``ParameterHandler`` as initialization for their data structures? Reducing dependencies is very important.